web/src/main/java/com/bazaarvoice/emodb/web/migrator/LocalRangeMigrator.java

```diff
@@ -43,7 +45,8 @@ public LocalRangeMigrator(MigratorTools migratorTools, LifeCycleRegistry lifecyc
         _failedRangeMigrations = metricRegistry.meter(MetricRegistry.name("bv.emodb.migrator", "Migrator", "failed-range-migrations"));
     }
 
-    public RangeScanUploaderResult migrate(final int taskId, ScanOptions options, final String placement, ScanRange range, int maxWritesPerSecond) throws IOException, InterruptedException {
+    public RangeScanUploaderResult migrate(final int taskId, ScanOptions options, final String placement, ScanRange range,
+                                           MigratorStatusDAO statusDAO, String migrationId) throws IOException, InterruptedException {
```

Comment:
This injection is creating a false dependency.  `MigratorStatusDAO` is only being used to get the one value, `maxWritesPerSecond`.  If the intent is for the value to be variable it would be better to do something like:

- Inject a `Provider<Integer>` which can dynamically return the write rate; this provider could just delegate to `MigratorStatusDAO`.
- Create an interface like `MigratorRateLimiter` with the one `getMaxWritesPerSecond(String)` method, have `MigratorStatusDAO` implement it, then bind and inject the interface.